devices/flash-zynq7000/flashdrv.c

```diff
@@ -572,20 +600,38 @@ static int flashdrv_init(unsigned int minor)
 	fdrv_common.buff = (void *)ADDR_OCRAM_HIGH;
 
 	res = qspi_init();
-	if (res < 0)
+	if (res < 0) {
 		return res;
+	}
 
 	res = flashdrv_cfiRead(&info->cfi);
-	if (res < 0)
+	if (res < 0) {
 		return res;
+	}
 
 	res = flashcfg_infoResolve(info);
-	if (res < 0)
+	if (res < 0) {
 		return res;
+	}
 
 	for (i = 0; i < info->cfi.regsCount; ++i) {
-		if (CFI_SIZE_SECTION(info->cfi.regs[i].size) > SIZE_OCRAM_HIGH)
+		if (CFI_SIZE_SECTION(info->cfi.regs[i].size) > SIZE_OCRAM_HIGH) {
 			return -EINVAL;
+		}
+	}
+
+	if (info->init != NULL) {
+		res = info->init(info);
+		if (res < 0) {
+			return res;
+		}
+	}
+
+	/* Require data RXed due to dummy cycles to be byte aligned. */
+	/* This requirement is in place to simplify implementation of read. */
+	if ((info->cmds[info->readCmd].dummyCyc == (u8)-1) ||
```

Comment:
No need to use UB here, it's even less characters to write this ;)

(-1 cast to unsigned is UB, because in C standard negative number representation is not defined - it doesn't have to be U2, so (u8)-1 == 0xff is not _always_ true, though it is in most cases)
```suggestion
	if ((info->cmds[info->readCmd].dummyCyc == 0xffu ||
```